.github/workflows/chip-tool-tests.yml

```diff
@@ -0,0 +1,141 @@
+  schedule:
+    - cron: 0 2 * * * # Every day at 02:00
+  workflow_dispatch: # Manually on demand
+  push:
+    branches:
+      - "*"
+
+# Cancel previous PR/branch runs when a new commit is pushed
+concurrency:
+  group: ${{ github.ref }}
+  cancel-in-progress: true
+
+env:
+  CONNECTEDHOMEIP_TAG: "master" # "v1.1-branch" is not containing our current status, so we use master till 1.2 comes out
+
+jobs:
+  chip-tests:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out matter.js
+        uses: actions/checkout@v3
+
+      - uses: dorny/paths-filter@v2
+        id: changes
+        with:
+          filters: |
+            src:
+              - "packages/matter-node.js/test-chip/**"
+              - "packages/matter.js/src/cluster/**"
+              - "packages/matter.js/src/tlv/**"
+              - ".github/workflows/chip-tool-tests.yml"
+
+      - if: ${{ github.event_name == 'schedule' || (github.event_name == 'push' && steps.changes.outputs.src == 'true') || github.event_name == 'workflow_dispatch' || contains(github.event.head_commit.message, '[execute-chiptests]') == true }}
+        name: Install Debian deps
+        id: install-deps
+        run: |
+          sudo apt-get update
+          sudo apt-get install git gcc g++ pkg-config libssl-dev libdbus-1-dev libglib2.0-dev libavahi-client-dev ninja-build python3-venv python3-dev python3-pip unzip libgirepository1.0-dev libcairo2-dev libreadline-dev
+
+      - if: ${{ github.event_name == 'schedule' || (github.event_name == 'push' && steps.changes.outputs.src == 'true') || github.event_name == 'workflow_dispatch' || contains(github.event.head_commit.message, '[execute-chiptests]') == true || contains(github.event.head_commit.message, '[rebuild-chip]') == true }}
+        name: Prepare Bin Directory
+        id: prepare_bin
+        run: |
+          mkdir bin
+
+      # Cache Bin files
+      - if: ${{ github.event_name == 'schedule' || (github.event_name == 'push' && steps.changes.outputs.src == 'true') || github.event_name == 'workflow_dispatch' || contains(github.event.head_commit.message, '[execute-chiptests]') == true || contains(github.event.head_commit.message, '[rebuild-chip]') == true }}
+        name: Cache Bin files
+        id: cache-bin
+        uses: actions/cache@v3
+        env:
+          cache-name: chip-bin
+        with:
+          path: ./bin
+          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ env.CONNECTEDHOMEIP_TAG }}
+
+      # Only build if cache is not hit or if the commit message contains '[rebuild-chip]'
+      - if: ${{ (github.event_name == 'schedule' || (github.event_name == 'push' && steps.changes.outputs.src == 'true') || github.event_name == 'workflow_dispatch' || contains(github.event.head_commit.message, '[execute-chiptests]') == true || contains(github.event.head_commit.message, '[rebuild-chip]') == true) && (steps.cache-bin.outputs.cache-hit != 'true' || contains(github.event.head_commit.message, '[rebuild-chip]') == true) }}
```

Comment:
Grr it's annoying to have to do this, in theory could we use the connectedhomeip/chip-cert-bins Docker image?  It looks like they only publish arm64 images but maybe we could ask for amd64 builds?  Would certainly be faster

Reply:
I also thought that but this cert bins is compiled somehow differently as it seems. And I was unsure if we are allowed to use them.

Reply:
And second reason is that we so have more control whoch versionn we use because it seems that certification-binsa are upgraded to new spec versions way before spec is publsihed some times ... all this brings more flexibility if we compile ourself